### PR TITLE
Add Shell to default template

### LINKED ISSUE #4522: Proposal: Introduce Shell as the default way of using/learning .NET MAUI (templates, documentation, etc.)
### Description

The Shell concept was introduced in Xamarin.Forms 4.0 and has been brought over to .NET MAUI and in my opinion helps standardize application structure, navigation, theming, and at the same time abstracts away complicated structure of Flyout and Tabs. Today, developers coming to Xamarin.Forms and .NET MAUI for the first time will have to make a decision if they should use Shell or not, which has rippling effects down the road. The default templates for Xamarin.Forms have mostly been updated to default to Shell, however Blank app remains a non-Shell based application, and the documentation separates out Shell into a different section, which means that you have to know to go look for it to go learn the concepts. I believe that developers will be more successful learning and using .NET MAUI if Shell is the default way of building applications, learning applications, and is the default template when running `dotnet new maui`.

Benefits:
* Standard way of learning app structure, navigation, theming, and more
* URI navigation & constructor injection simplify development
* Integrated search handling
* Documentation can be simplified by focusing on Shell concepts first
* This is our chance to do it right and not make things confusing with multiple choices, make Shell the way.

I am not asking Shell to be required in .NET MAUI, so non-Shell based XF apps can easily migrated over and traditional navigation can still work just fine. But, I believe that Shell was created to make cross-platform development easier and from my personal usage it does.

Here is my PR for proposal changes to templates -> https://github.com/dotnet/maui/pull/4524

### Public API Changes

None, we would update the templates to include a Shell with theming and add a single page.

### Intended Use-Case

Make app development with .NET MAUI easier than ever, while still be backwards compatible.

### PR DESCRIPTION
**Pull request by jamesmontemagno**
This PR would fix #4522 

1.  Add shell into default template
2. Apply light/dark theming correctly
3. Change from Grid to VerticalStackLayout

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works. 

sample app of before and after shell https://github.com/jamesmontemagno/maui-4335-bug

Profiling.... run in release on a pixel 2 XL with the following settings:

```
<PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
		<RunAOTCompilation Condition="'$(TargetFramework)' == 'net6.0-android'">true</RunAOTCompilation>
		<AndroidEnableProfiledAot Condition="'$(TargetFramework)' == 'net6.0-android'">true</AndroidEnableProfiledAot>
	</PropertyGroup>
```
**Non-Shell (file new)**
Average(ms): 1386.1
Std Err(ms): 2.06801031589948
Std Dev(ms): 6.53962282296668

**Shell (file new and shell)**
Average(ms): 1777.8
Std Err(ms): 3.32264954516723
Std Dev(ms): 10.507140429251

**Shell with a custom profile**
Peppers built me a custom profile, since the one that is built by default is not shell. With this enabled in my sample project we get the following results:
Average(ms): 1782.8
Std Err(ms): 2.9128069699938
Std Dev(ms): 9.21110440959413

### CODE REVIEW

File: src/Templates/src/templates/maui-mobile/AppShell.xaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            <Style BasedOn="{StaticResource BaseStyle}" TargetType="FlyoutItem" />
+        </ResourceDictionary>
+    </Shell.Resources>
+
+
+
+    <ShellContent
+        Title="Hello, World!"
+        ContentTemplate="{DataTemplate local:MainPage}"
+        Route="MainPage" />
```
Comment (jonathanpeppers) [inline src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: Hey, I wanted to bring up. What is the impact of startup time for these changes to the template?

I'd be curious of the average of 10 runs on a device using the script here:

https://github.com/jonathanpeppers/maui-profiling/blob/main/scripts/profile.ps1

Comparing a regular release build would be fine.

We should do this if it is better, but maybe we also want some buy-in from the MAUI team to improve any startup cost when adding Shell by default?

Comment (jamesmontemagno) [reply src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: Ideally it is no impact... as that should be the goal of shell for sure.

Comment (jamesmontemagno) [reply src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: Non-Shell
Average(ms): 2682.8
Std Err(ms): 18.5159390796146
Std Dev(ms): 58.552540508504

Shell:
Average(ms): 3489.2
Std Err(ms): 45.70285670624
Std Dev(ms): 144.52512276802

Comment (jonathanpeppers) [reply src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: I saw some things in `dotnet trace` like this line taking up 20ms:

https://github.com/dotnet/maui/blob/303e3ef5e084419ed94a0b7a93171479711c92fa/src/Controls/src/Core/Shell/ShellUriHandler.cs#L243

I bet your numbers would be a bit better on a device. Is it also a Release build?

Comment (jamesmontemagno) [reply src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: running on device now

Comment (jamesmontemagno) [reply src/Templates/src/templates/maui-mobile/AppShell.xaml:33]: Added details

**Comment by rachelkang** [comment]
Logged a couple of related bugs we'll need to work through:

* https://github.com/dotnet/maui/issues/5105
* https://github.com/dotnet/maui/issues/5106
* https://github.com/dotnet/maui/issues/5107

**Review by PureWeen — APPROVED** [review]

**Comment by Beelzenef** [comment]
congrats for the merge, @jamesmontemagno! 😀

**Pull request merged**